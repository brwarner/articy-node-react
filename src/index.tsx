export { EntityAvatar } from './components/EntityAvatar';
export { ErrorMessage } from './components/ErrorMessage';
export { FlowText } from './components/FlowText';
export { ImageAsset } from './components/ImageAsset';
export { VariableDebugView } from './components/VariableDebugView';

export * from './DatabaseContext';
export * from './formatter';
export * from './util';
export * from './branches';
