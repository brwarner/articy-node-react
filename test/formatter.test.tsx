import { formatDisplayText, FormatProcessor } from '../src/formatter';
import renderer from 'react-test-renderer';
import React from 'react';

const testKey = 'test-format';

function render(text: string, processor?: FormatProcessor) {
  return renderer.create(formatDisplayText(testKey, text, processor)).toJSON();
}

test('No paragraphs', () => {
  expect(render('This is some text.')).toMatchSnapshot();
});

test('Basic paragraphs', () => {
  expect(render('One paragraph.\n\nTwo paragraph')).toMatchSnapshot();
});

test('No paragraphs for single newlines', () => {
  expect(render('This is all\nOne paragraph.')).toMatchSnapshot();
});

test('Bold and italics', () => {
  expect(
    render('Some text in <b>bold</b> vs <i>italics</i>.')
  ).toMatchSnapshot();
});

test('Three newlines should break a paragraph then trim', () => {
  expect(
    render('One paragraph.\n\n\nStart of new trimmed paragraph.')
  ).toMatchSnapshot();
});

test('Bold across paragraph break', () => {
  expect(
    render('What happens if we <b>bold across\n\nparagraphs</b>.')
  ).toMatchSnapshot();
});

const replaceNo: FormatProcessor = function(fragment) {
  const processed = fragment
    .split('no')
    .flatMap((v, i) => [v, <input key={i.toString()} />]);
  processed.pop();
  return processed;
};

test("Processor to replace 'no' with <input>", () => {
  expect(
    render(
      'You should see no no in here.\n\nAnd no no in here.\n\nno',
      replaceNo
    )
  ).toMatchSnapshot();
});
