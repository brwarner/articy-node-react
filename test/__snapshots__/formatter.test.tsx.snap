// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic paragraphs 1`] = `
Array [
  <p>
    One paragraph.
  </p>,
  <p>
    Two paragraph
  </p>,
]
`;

exports[`Bold across paragraph break 1`] = `
Array [
  <p>
    What happens if we &lt;b&gt;bold across
  </p>,
  <p>
    paragraphs&lt;/b&gt;.
  </p>,
]
`;

exports[`Bold and italics 1`] = `
<p>
  Some text in 
  <strong>
    bold
  </strong>
   vs 
  <i>
    italics
  </i>
  .
</p>
`;

exports[`Bold in italics 1`] = `
<p>
  Here are some 
  <i>
    <strong>
      bold in italics
    </strong>
  </i>
</p>
`;

exports[`Bold in italics with edges 1`] = `
<p>
  Here are some 
  <i>
    edged 
    <strong>
      bold in italics
    </strong>
     edged
  </i>
</p>
`;

exports[`Italics in bold 1`] = `
<p>
  Here are some 
  <strong>
    <i>
      italics in bold
    </i>
  </strong>
</p>
`;

exports[`No paragraphs 1`] = `
<p>
  This is some text.
</p>
`;

exports[`No paragraphs for single newlines 1`] = `
<p>
  This is all
  <br />
  One paragraph.
</p>
`;

exports[`Processor to replace 'no' with <input> 1`] = `
Array [
  <p>
    You should see 
    <input />
     
    <input />
     in here.
  </p>,
  <p>
    And 
    <input />
     
    <input />
     in here.
  </p>,
  <p>
    
    <input />
    
  </p>,
]
`;

exports[`Three newlines should break a paragraph then trim 1`] = `
Array [
  <p>
    One paragraph.
  </p>,
  <p>
    Start of new trimmed paragraph.
  </p>,
]
`;
